test/e2e/mock-e2e.js

```diff
@@ -190,6 +190,54 @@ async function setupMocking(server, testSpecificMock, { chainId }) {
       };
     });
 
+  // Both are added to support swaps e2e tests
```

Comment:
Gas prices are fetch within [swaps UI](https://github.com/MetaMask/metamask-extension/blob/e2ff620d606b6cfc733847c87ebb8a0bc6201c08/ui/pages/swaps/swaps.util.ts#L248)

I am not sure why chainId is override [in the tests](https://github.com/MetaMask/metamask-extension/blob/e2ff620d606b6cfc733847c87ebb8a0bc6201c08/shared/lib/swaps-utils.js#L147)

Basically if we don't put these mocks then swaps e2e tests will fail, because it's not being caught by XHR mocks and auth modal will appear (as expected in the screenshot)

This is a quick fix but just patching the current solution, fyi

![Screenshot 2024-03-28 at 17 47 19](https://github.com/MetaMask/metamask-extension/assets/7644512/44deb15a-97d3-4dff-8ae7-ee63428f1e82)

Reply:
To clarify, are you saying you believe this wasn't being mocked previously and was hitting the real endpoint?

Reply:
Indeed, this issue went unnoticed previously because the former endpoint shared the same domain, allowing it to bypass the privacy snapshot. However, the new endpoint is protected by authentication, which explains why it is prompting for credentials, as in the screenshot.